pontoon/base/placeables.py

```diff
@@ -0,0 +1,45 @@
+import re
+
+
+class PythonFormatNamedString:
+    parser = r"(%\([[\w\d!.,[\]%:$<>+\-= ]*\)[+|-|0\d+|#]?[.\d+]?[s|d|e|f|g|o|x|c|%])"
+
+
+class PythonFormatString:
+    parser = r"(\{{?[\w\d!.,[\]%:$<>+-= ]*\}?})"
+
+
+class PythonFormattingVariable:
+    parser = r"(%(%|(\([^)]+\)){0,1}[-+0#]{0,1}(\d+|\*){0,1}(\.(\d+|\*)){0,1}[hlL]{0,1}[diouxXeEfFgGcrs]{1}))"
+    match_index = 0
+
+
+class FluentTerm:
+    parser = r"({ ?-[^}]* ?})"
```

Comment:
```suggestion
    parser = r"({ *-[^}]*})"
```

The spec allows for arbitrary amounts of whitespace, not up to one character. Also, the `[^}]` captures whitespace as well as any valid parameters, so FluentParametrizedTerm form is not required.